packages/block-editor/src/components/gradients/index.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * External dependencies
+ */
+import { find } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback } from '@wordpress/element';
+import { useSelect } from '@wordpress/data';
+
+export function __experimentalGetGradientClass( gradientSlug ) {
+	if ( ! gradientSlug ) {
+		return undefined;
+	}
+	return `has-${ gradientSlug }-gradient-background`;
+}
+
+function getGradientValueBySlug( gradients, slug ) {
+	const gradient = find( gradients, [ 'slug', slug ] );
+	return gradient && gradient.gradient;
+}
+
+function getGradientSlugByValue( gradients, value ) {
+	const gradient = find( gradients, [ 'gradient', value ] );
+	return gradient && gradient.slug;
+}
+
+export function __experimentalUseGradient( attributes, setAttributes ) {
```

Comment:
I think we can access the block client id from the block edit context and avoid `attributes` and `setAttributes`  arguments here.